src/P/Either.hs

```diff
@@ -24,3 +34,11 @@ ecase e l = flip (either l) e
 
 flipEither :: Either a b -> Either b a
 flipEither = either Right Left
+
+lefts :: (Foldable f) => f (Either a b) -> [a]
```

Comment:
Should we just add `partionEithers`? I would find that more safe than always discarding one side of the other. The choice can then be up to the consumer to discard something and make it more obvious.

Reply:
I didn't know it existed. I've added `partitionEithers`, keeping lefts+rights for convenience.